### PR TITLE
Get prioritized run ids

### PR DESCRIPTION
**Pull request by jmsanders**
Given the sorting algorithm described in:

https://docs.dagster.io/guides/customizing-run-queue-priority

Postgres has a slightly different path extraction implementation than MySQL and sqlite have.

Shifting this prioritization down to the db unblocks a refactor to the queued run coordinator daemon to prevent needing to load every queued run into memory.

### CODE REVIEW

File: python_modules/dagster/dagster/_core/instance/__init__.py [hunk 1]
```diff
@@ -1735,8 +1735,14 @@ def get_run_ids(
         filters: Optional[RunsFilter] = None,
         cursor: Optional[str] = None,
         limit: Optional[int] = None,
+        prioritized: bool = False,
     ) -> Sequence[str]:
-        return self._run_storage.get_run_ids(filters, cursor=cursor, limit=limit)
+        return self._run_storage.get_run_ids(
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/_core/instance/__init__.py:1740]: `cursor` won't work with `prioritized` right? probably worth actively erroring on non-None `cursor` and `prioritized` `True`

**Review by alangenfeld — APPROVED** [review]
bool flag isn't my favorite but probably the most reasonable way forward here

File: python_modules/dagster/dagster/_core/instance/__init__.py (hunk 1 above)
Comment (jmsanders) [reply python_modules/dagster/dagster/_core/instance/__init__.py:1740]: Hm. I think it should still work insofar as it'll only prioritize runs after that cursor. I think you're right that it's unlikely anybody actually wants to call it that way.

**Comment by jmsanders** [comment]
Closed in favor of https://github.com/dagster-io/dagster/pull/18347

**Pull request closed without merging**